### DISCUSSION TITLE
Original error was: No module named 'numpy.core._multiarray_umath'

### DISCUSSION THREAD

**Discussion by lammps-vasp** [deepmodeling/deepmd-kit#618] [Q&A]
The machine time version is python3.7.3 numpy1.20.3; If the python version path is commented out, there will still be conflicts. Re install the package or it cannot be installed normally. The following is the error code. What should I do

caep003@ln06 ~/DP_ZHANG/soft> dp -h
Traceback (most recent call last):
  File "/public/home/users/zkchu/tools/python-3.7.3/lib/python3.7/site-packages/numpy/core/__init__.py", line 22, in <module>
    from . import multiarray
  File "/public/home/users/zkchu/tools/python-3.7.3/lib/python3.7/site-packages/numpy/core/multiarray.py", line 12, in <module>
    from . import overrides
  File "/public/home/users/zkchu/tools/python-3.7.3/lib/python3.7/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/dp", line 7, in <module>
    from deepmd.main import main
  File "/public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/python3.8/site-packages/deepmd/__init__.py", line 1, in <module>
    from .env import set_mkl
  File "/public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/python3.8/site-packages/deepmd/env.py", line 4, in <module>
    import numpy as np
  File "/public/home/users/zkchu/tools/python-3.7.3/lib/python3.7/site-packages/numpy/__init__.py", line 145, in <module>
    from . import core
  File "/public/home/users/zkchu/tools/python-3.7.3/lib/python3.7/site-packages/numpy/core/__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.8 from "/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/python"
  * The NumPy version is: "1.20.3"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

**Comment by AnguseZhang** [comment]
Did you installed DeePMD-kit with offline installation package?  If it's true, there's already an executable python attached with all relevant python packages such as numpy, which means you don't need to install other packages and should make sure there is no conflicts bewteen environments (there seems confusing since you both involve `python3.7` and `python3.8`).

**Comment by lammps-vasp** [comment]
This kind of situation is sudden, there is no such problem before. It's the offline package installed. Python3.8 is available after installing deepmd offline package. I set up a virtual environment to install offline package to avoid environment conflict, but I still report the same error


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年5月12日(星期三) 下午5:09
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] Original error was: No module named 'numpy.core._multiarray_umath' (#618)





 
Did you installed DeePMD-kit with offline installation package?  If it's true, there's already an executable python attached with all relevant python packages such as numpy, which means you don't need to install other packages and should make sure there is no conflicts bewteen environments (there seems confusing since you both involve python3.7 and python3.8).
